### ISSUE TITLE
Retry Image Src Loading

### ISSUE THREAD

**Issue by sbowler** [openlayers/openlayers#6292]
Hey guys, thanks for all of your hard work on this and responsiveness to issues and questions. I think this is basically a feature request for a feature that I think was in OL2 and I think it's probably not too hard to implement something, but I'm just not that familiar with your large code base.

Anyways, I was wondering if you could add the ability to retry loading an image source multiple times if failing to load. Currently we are using `ol.source.ImageStatic` to add geo-rectified PNGs to the map, but sometimes they are not created yet when added. So we have manually check for them and wait for them to be created ourselves prior to adding the source or obviously we could reload the src at that point, either way it's a pain.

So basically I think it would be an awesome feature if we could pass in something like retry the source 3 times on failure (like 404 or something like that) and maybe wait 5 or 10 seconds in between attempts. Ideally I think both of those numbers would be configurable options you pass in and would be 0/disabled by default. 

Any thoughts? Thanks again guys.

**Comment by tschaub** [comment]
I think `retries` (`number`, defaults to 2) and `retryDelay` (`number`, defaults to 0) would make for nice source options.

**Comment by tschaub** [comment]
Could also make sense to have `retries` default to 0 (I've got a couple use cases where some 4xx is expected).

**Comment by sbowler** [comment]
I could see either way I guess. A default of 0 would work fine for me. A lot of tiles and images are pre-generated or generated on the fly in most cases I would think and that's true for me. I just have one case where it's generated on demand that this would be very useful.

**Comment by Morgul** [comment]
I like the idea of adding a `retries` option, but so far my solution to similar problems has just been writing a custom `tileLoadFunction`. Here's a contrived example of timing out the request after 1s:

```javascript
tileLoadFunction(imageTile, url)
    {
        var img = imageTile.getImage();

        var coords = imageTile.getTileCoord();
        coords = {x: coords[1], y: coords[2], z: coords[0]};
        coords.y = Math.pow(2, coords.z) + coords.y;

        var handle = setTimeout(() =>
            {
                console.warn('Tile timed out:', url);
                img.alt = "Tile timed out while loading";
                img.src = this.options.placeholder || "";
            }, 1000);

        $http.get('/some/special/image'), (error, response) =>
            {
                clearTimeout(handle);

                if(error)
                {
                    img.alt = JSON.stringify(error);
                    img.src = "";

                    return console.error("Error while getting tile!", error);
                } // end if

                // Convert our binary data to a blob
                var blob = new Blob([ response.data ], { type: response.contentType });
                img.src = window.URL.createObjectURL(blob);
            });
    } // end tileLoadFunction
```

It shouldn't be hard to turn my example into something with retry support. (And the code can probably be cleaned up a little.)
